### PR TITLE
[PM-3285] Autofill v2 Feature Branch

### PR DESCRIPTION
**Pull request by cagonzalezcs**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This branch and PR are being used to route work done for the re-write of the core autofill content script, the unit tests created for the autofill service, and the feature flag implementation created for autofill v2.

The review of those changes will occur and will be merged into this PR before we provide final approvals and merge the overall work into master.

### CODE REVIEW

**Comment by cagonzalezcs** [comment]
@jprusik @MGibson1 

This work has been thoroughly regression tested by QA and is ready for final approvals to merge into master. Work done for this ticket is representative of the previously approved work merged in from #5371 , #5453  and #5695

The only significant code changes that have occurred after approval and merging of those three PRs is a console log that was added to the `AutofillService` class identifying which version of autofill.js was loading. This change has been reverted following QA's regression.

File: apps/browser/package.json [hunk 1]
```diff
@@ -6,7 +6,6 @@
     "build:mv3": "cross-env MANIFEST_VERSION=3 webpack",
     "build:watch": "webpack --watch",
     "build:watch:mv3": "cross-env MANIFEST_VERSION=3 webpack --watch",
-    "build:watch:autofill": "cross-env AUTOFILL_VERSION=2 webpack --watch",
```
Comment (jprusik) [inline apps/browser/package.json:9]: 🎉

**Review by jprusik — DISMISSED** [review]

**Review by justindbaur — APPROVED** [review]
Just one comment about permissions that we will need in MV3 but since that manifest file isn't used yet, this looks good from Platform.

File: apps/browser/src/platform/browser/browser-api.ts [hunk 2]
```diff
@@ -308,4 +308,31 @@ export class BrowserApi {
...
+  /**
+   * Extension API helper method used to execute a script in a tab.
+   * @see https://developer.chrome.com/docs/extensions/reference/tabs/#method-executeScript
+   * @param {number} tabId
+   * @param {chrome.tabs.InjectDetails} details
+   * @returns {Promise<unknown>}
+   */
+  static executeScriptInTab(tabId: number, details: chrome.tabs.InjectDetails) {
+    if (BrowserApi.manifestVersion === 3) {
+      return chrome.scripting.executeScript({
```
[the remark this answers is not in the document]
Comment (cagonzalezcs) [reply apps/browser/src/platform/browser/browser-api.ts:321]: @justindbaur Yes we will, I'll ensure that it is added within the scope of this work: https://github.com/bitwarden/clients/pull/6141

I'll make sure to document the change as well. There are other issues with Manifest v3, of course, so further investigation on that front needs to occur. 

Thank you so much for the review!

**Review by jprusik — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Review by differsthecat — APPROVED** [review]
Great work on this everyone! 🎉 

I looked over all commits not associated with a PR and they look good, and the high-level flow of things and it all looks great!

**Comment by RafaelKr** [comment]
Congratulations on getting this merged! 🥳
I'm very excited to "see" this in action after the next release!

**Pull request merged**